airflow/migrations/versions/0029_3_0_0_remove_is_subdag.py

```diff
@@ -40,37 +41,23 @@
 airflow_version = "3.0.0"
 
 
-def _column_exists(inspector, column_name):
-    return column_name in [col["name"] for col in inspector.get_columns("dag")]
-
-
-def _index_exists(inspector, index_name):
-    return index_name in [index["name"] for index in inspector.get_indexes("dag")]
-
-
 def upgrade():
     """Remove ``is_subdag`` column from DAGs table."""
-    conn = op.get_bind()
-    inspector = sa.inspect(conn)
-
-    with op.batch_alter_table("dag", schema=None) as batch_op:
-        if _index_exists(inspector, "idx_root_dag_id"):
-            batch_op.drop_index("idx_root_dag_id")
-        if _column_exists(inspector, "is_subdag"):
-            batch_op.drop_column("is_subdag")
-        if _column_exists(inspector, "root_dag_id"):
-            batch_op.drop_column("root_dag_id")
+    dialect = op.get_bind().dialect.name
+    with op.batch_alter_table("dag") as batch_op:
+        batch_op.drop_column("is_subdag")
+        batch_op.drop_column("root_dag_id")
+        if op.get_bind().dialect.name == "mysql":
+            mysql_drop_index_if_exists("idx_root_dag_id", "dag", op)
```

Comment:
Do we need the if exists check here? Shouldn't it always exist when upgrading? (i.e. was this needed, or was the original Pr author being overly defensive?)

Reply:
The latter -- I think it was a remnant of me doing a lot of upgrades & downgrades -- and I wanted it to be idempotent for that. The original author being me :D

Reply:
Done